sphinx/domains/__init__.py

```diff
@@ -404,3 +409,95 @@ def get_enumerable_node_type(self, node: Node) -> str | None:
     def get_full_qualified_name(self, node: Element) -> str | None:
         """Return full qualified name for given node."""
         pass
+
+    def intersphinx_add_entries_v2(self, store: Any,
+                                   data: dict[str, dict[str, Any]]) -> None:
+        """Store the given *data* for later intersphinx reference resolution.
+
+        This method is called at most once with all data loaded from inventories in
+        v1 and v2 format.
+
+        The *data* is a dictionary indexed by **object type**, i.e, a key from
+        :attr:`object_types`. The value is a dictionary indexed by **object name**,
+        i.e., the **name** part of each tuple returned by :meth:`get_objects`.
```

Comment:
I'd suggest adding an example (the structure would be clearer IMO), e.g.: 

```python
data = {
	'class': {'pkg.mod.Foo': SomeIntersphinxData(...)},
	'method': {'pkg.mod.Foo.bar': SomeIntersphinxData(...)},
}
```

Reply:
Good idea. Done.